services/web/Dockerfile

```diff
@@ -15,14 +16,27 @@ WORKDIR /service
 
 # Install dependencies and store yarn cache
 COPY package.json yarn.lock ./
-RUN --mount=type=cache,target=/root/.yarn YARN_CACHE_FOLDER=/root/.yarn yarn install --frozen-lockfile
```

Comment:
@kaareal Wasn't this cache added to speed up build time?

Reply:
yeah just realized it moments ago when comparing build times need to do some changes to fix this.

Reply:
not sure if really does for this, yarn install but it does for the yarn build for sure. perhaps @andrewplummer knows more. I am gonna leave it in for now.

Reply:
Official word from ChatGPT :)

> 
The command you provided in a Dockerfile, "RUN --mount=type=cache,target=/root/.yarn," utilizes the mount feature of the Docker build process.

When the Dockerfile is being built, this specific command instructs Docker to create a cache mount at the specified target directory, "/root/.yarn." The cache type used here is not specified, so it could be a local cache or a remote cache.

Mounting a cache in this manner allows Docker to persist data between different build stages or runs of the Dockerfile. In this case, it seems to be related to Yarn, a package manager commonly used in JavaScript development.

By using a cache mount, Docker can avoid redundant downloads or installations of dependencies during the build process. Instead, it can use the cached data stored at "/root/.yarn" to speed up subsequent builds or runs, as long as the cache remains valid and up to date.

Reply:
right so if understand this correctly, adding a package should just take the time to install that one package instead of full build.

Reply:
Ya that's the idea.